call_host.sh

```diff
@@ -8,15 +9,70 @@ if [ -f "$CALL_HOST_CONFIG" ]; then
 	source "$CALL_HOST_CONFIG"
 fi
 
+# zsh / bash compatibility helpers
+is_zsh(){
+	# detect the current shell process name (portable ps usage)
+	if command -v ps >/dev/null 2>&1; then
+		# get last path component if ps returns full path
+		p="$(ps -p $$ -o comm= 2>/dev/null | awk -F/ '{print $NF}')"
+		case "$p" in
+			zsh) return 0 ;;
```

Comment:
because there is only a `zsh` case here, in the `bash` case, this will proceed through the fallback below before returning non-zero. I would prefer stopping here if `bash` is directly detected.

Reply:
OK, will add

```
bash) return 1 ;;
```